src/main/java/core/basesyntax/servise/DataConverterImpl.java

```diff
@@ -0,0 +1,38 @@
+package core.basesyntax.servise;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.Operation;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    private static final String COMMA = ",";
+    private static final int OPERATION_INDEX = 0;
+    private static final int FRUIT_INDEX = 1;
+    private static final int QUANTITY_INDEX = 2;
+    private static final int REQUIRED_PARTS_LENGTH = 3;
+
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> content) {
```

Comment:
Follow SOLID rules. This method seems to be doing multiple things: it's parsing the lines, validating the format, and then creating FruitTransaction objects. Consider refactoring to separate these concerns for better adherence to the Single Responsibility Principle.